Xamarin.Forms.Core.UITests.Shared/Utilities/Gestures.cs

```diff
@@ -36,8 +54,15 @@ public static bool ScrollForElement (this IApp app, string query, Drag drag, int
 			}
 
 			if (count != maxSteps) {
+				var elements = app.Query(elementQuery);
+				while (elements.Count() == 0)
+				{
+					app.WaitForElement(elementQuery);
+					elements = app.Query(elementQuery);
+				}
```

Comment:
Might this result in a hang if the count continues to return zero for some reason? Might we consider refactoring `GetAppRect` to `QueryNTimes` so it could be used here in addition to the other callsites?